app/models/html_document.rb

```diff
@@ -18,8 +18,10 @@ def keywords
   # Returns client-side redirect url
   def redirect_url
     refresh = html.css('meta[http-equiv]').detect{|node| /refresh/i === node['http-equiv'] }
-    new_path = refresh.try(:[],'content')&.gsub(/.*url=/i,'')
-    URI(url).merge(URI(new_path)).to_s if new_path
+    if refresh
+      new_path = refresh['content'].match(/.*URL=('|")?(?<path>[^('|")]*)/i)[:path]
```

Comment:
I think this regex is more complicated than it needs to be, and the character class towards the end that is meant to only exclude single and double quotes is also excluding parentheses and pipes. Parentheses and pipes inside a character class lose their special meaning. Here you can see the regex not capturing the full path:

```
irb(main):001:0> "URL='./new().html'".match(/.*URL=('|")?(?<path>[^('|")]*)/i)[:path]
=> "./new"
```

Using just a character class and no groups is simpler and more correct:

```
irb(main):002:0> "URL='./new().html'".match(/.*URL=['"]?(?<path>[^'"]*)/i)[:path]
=> "./new().html"
```

It would be good to see some additional specs to show that the regex is doing what's intended.

Reply:
You have revealed my ignorance; I thought `(a|b`) meant `'a' or 'b'`, not `'a' or '|' or 'b'`. I will make those changes, then check "Regexes for Dummies" out of the library.